foca/api/register_openapi.py

```diff
@@ -0,0 +1,180 @@
+                https://swagger.io/specification/#oas-document
+        add_operation_fields: Fields to be added/modified in Operation Objects.
+                By making use of this dictionary object, the user can add or
+                modify specific fields of the Operation Objects of each Path
+                Info Object fo the required paths field; this is mostly to
+                support the addition (or replacement) of the
+                x-swagger-router-controller field
+                For OpenAPI 2, see
+                https://swagger.io/specification/v2/#operation-object
+                For OpenAPI 3, see
+                https://swagger.io/specification/#operation-object
+
+        An example OpenAPIConfig instance would look like this:
+            my_specs_config = OpenAPIConfig(
+                out_file='/path/to/my/modified/specs.yaml',
+                append=[
+                    {
+                        'security':
+                            'jwt':
+                                'type': 'apiKey',
+                                'name': 'Authorization',
+                                'in': 'header',
+                    },
+                    {
+                       'my_other_root_field': 'some_value',
+                    },
+                ],
+                add_operation_fields = {
+                    'x-swagger-router-controller': 'controllers.ga4gh.wes',
+                    'some-other-custom-field': 'some_value',
+                },
+            )
+    """
+
+    def __init__(
+            self,
+            out_file: Optional[str],
+            append: Optional[List[Dict]],
+            add_operation_fields: Optional[Dict]
+    ) -> None:
+        super().__init__()
+        if out_file:
+            self.out_file = out_file
+        if append:
+            self.append = append
+        if add_operation_fields:
+            self.add_operation_fields = add_operation_fields
+
+
+def register_openapi(
+        app: App,
+        specs: Dict[str, OpenAPIConfig]
+) -> App:
+    """
+    Registers OpenAPI specs with Connexion app
+
+    Args:
+        app: A Connexion app instance
+        specs: A Dictionary containing the absolute path of OpenAPI
+            specification files as keys and OpenAPIConfig objects as
+            values for further additions/modifications to the configuration.
+
+    Returns:
+        A Connexion app instance
+
+    Raises:
+        InvalidSpecification: The InvalidSpecification exception is
+                            raised in case the configuration file is
+                            malformed (invalid YAML/JSON formats) or
+                            in the case of invalid specs field access.
+        FileNotFoundError: Any of the files were not found.
+        PermissionError: Any of the files were not accessible.
+    """
+    # Iterate over list of API specs
+    for spec_path, spec in specs:
+        with open(spec_path, 'r') as spec_file:

```

Comment:
To test this code, you will need to write some fake config files, a valid OpenAPI2 and OpenAPI3 one perhaps, and some faulty one(s). I'm sure you can find minimal examples for OpenAPI2/3 online

Reply:
You can take the petstore examples from here (available both for OpenAPI 2.x and 3.x): https://github.com/OAI/OpenAPI-Specification/tree/master/examples